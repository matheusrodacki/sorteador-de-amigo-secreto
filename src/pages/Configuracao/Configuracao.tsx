import Card from '../../components/Card';
import Formulario from '../../components/Formulario/Formulario';
import ListaParticipantes from '../../components/ListaParticipantes/ListaParticipantes';
import Rodape from '../../components/Rodape/Rodape';

const Configuracao = () => {
  return (
    <Card>
      <section>
        <Formulario />
        <ListaParticipantes />
        <Rodape />
      </section>
    </Card>
  );
};
export default Configuracao;
