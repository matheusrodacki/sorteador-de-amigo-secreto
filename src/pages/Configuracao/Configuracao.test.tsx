import { render } from '@testing-library/react';
import { RecoilRoot } from 'recoil';
import Configuracao from './Configuracao';

const mockNavegacao = jest.fn();

jest.mock('react-router-dom', () => {
  return {
    useNavigate: () => mockNavegacao,
  };
});

describe('a página de configuração', () => {
  test('deve ser renderizada corretamente', () => {
    // implemente o teste aqui
    const { container } = render(
      <RecoilRoot>
        <Configuracao />
      </RecoilRoot>
    );
    expect(container).toMatchSnapshot();
  });

  test('deve ter um título', () => {
    // implemente o teste aqui
  });
  test('deve ter um campo para inserir o nome do participante', () => {
    // implemente o teste aqui
  });
  test('deve ter um botão para adicionar o participante', () => {
    // implemente o teste aqui
  });
  test('deve ter uma lista de participantes', () => {
    // implemente o teste aqui
  });
});
