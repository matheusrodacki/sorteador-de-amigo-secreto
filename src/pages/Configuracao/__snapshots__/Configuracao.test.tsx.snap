// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a página de configuração deve ser renderizada corretamente 1`] = `
<div>
  <div
    class="card"
  >
    <section>
      <form>
        <div
          class="grupo-input-btn"
        >
          <input
            placeholder="Insira os nomes dos participantes"
            type="text"
            value=""
          />
          <button
            disabled=""
          >
            Adicionar
          </button>
          
        </div>
      </form>
      <ul />
      <footer
        class="rodape-configuracoes"
      >
        <button
          class="botao"
          disabled=""
        >
          Iniciar Brincadeira
        </button>
      </footer>
    </section>
  </div>
</div>
`;
